### PR TITLE
dev/core#1344 Re-work if statements and remove excess ifs

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This removes some excess ifs and re-works the order of the if logic as per #156764 this is a split off from #15651

Before
----------------------------------------
If logic complicated

After
----------------------------------------
If logic simplified

ping @eileenmcnaughton

### CODE REVIEW

File: xml/templates/message_templates/membership_online_receipt_text.tpl [hunk 1]
```diff
@@ -170,8 +163,14 @@
 {$address}
 
 {$email}
-{/if} {* End ! is_pay_later condition. *}
-{/if}
+{if $contributeMode eq 'direct' AND !$is_pay_later AND ( $amount GT 0 OR $membership_amount GT 0 ) }
```
Comment (eileenmcnaughton) [inline xml/templates/message_templates/membership_online_receipt_text.tpl]: This if can go

Comment (eileenmcnaughton) [reply xml/templates/message_templates/membership_online_receipt_text.tpl]: Maybe after the other is merged because I forsee a conflict below

Comment (seamuslee001) [reply xml/templates/message_templates/membership_online_receipt_text.tpl]: @eileenmcnaughton it will in #15651

**Comment by seamuslee001** [comment]
@eileenmcnaughton i have now rebased this after #15651 has been merged

**Comment by eileenmcnaughton** [comment]
Looks good to me!

File: xml/templates/message_templates/contribution_online_receipt_html.tpl [hunk 2]
```diff
@@ -311,32 +311,30 @@
       </tr>
      {/if}
 
-     {if ! ($contributeMode eq 'notify' OR $contributeMode eq 'directIPN') and $is_monetary}
-      {if $is_pay_later && !$isBillingAddressRequiredForPayLater}
```
Comment (eileenmcnaughton) [inline xml/templates/message_templates/contribution_online_receipt_html.tpl]: Dang - this template is the one that needs a test change

**Comment by seamuslee001** [comment]
@eileenmcnaughton have pulled out the changes to that template

**Comment by eileenmcnaughton** [comment]
Cool getting close now - only one tpl left here & only one search jcalendar!

**Comment by seamuslee001** [comment]
Test fail unrelated

**Pull request merged**